ui/browsing/albumpage.go

```diff
@@ -202,7 +202,13 @@ func NewAlbumPageHeader(page *AlbumPage) *AlbumPageHeader {
 						sharedutil.TracksToIDs(a.page.tracks))
 				}),
 				fyne.NewMenuItem("Download", func() {
-					a.page.contr.ShowDownloadDialog(a.page.tracks)
+					go func() {
+						album, err := a.page.mp.GetAlbum(a.albumID)
```

Comment:
Rather than calling the server again to get the album details, you can add a `title` field to the AlbumPageHeader struct that saves the title during the Update() call (or you could just reach into the titleLabel RichText to get it).

Once that's done there's no need for an async goroutine here since you won't be making any network calls

Reply:
I like this idea, I wasn't sure about the approach that I implemented. Thanks, I'll change it!